go/libkb/json.go

```diff
@@ -51,27 +51,37 @@ func (f *JSONFile) GetWrapper() *jsonw.Wrapper {
 func (f *JSONFile) Exists() bool { return f.exists }
 
 func (f *JSONFile) Load(warnOnNotFound bool) error {
+	found, err := f.LoadCheckFound()
+	if err != nil {
+		return err
+	}
+	if !found {
+		msg := fmt.Sprintf("No %q file found; tried %s", f.which, f.filename)
+		if warnOnNotFound {
+			f.G().Log.Warning(msg)
+		} else {
+			f.G().Log.Debug(msg)
+		}
+	}
+	return nil
+}
+
+func (f *JSONFile) LoadCheckFound() (found bool, err error) {
 	f.G().Log.Debug("+ loading %q file: %s", f.which, f.filename)
 	file, err := os.Open(f.filename)
 	if err != nil {
 		if os.IsNotExist(err) {
-			msg := fmt.Sprintf("No %q file found; tried %s", f.which, f.filename)
-			if warnOnNotFound {
-				f.G().Log.Warning(msg)
-			} else {
-				f.G().Log.Debug(msg)
-			}
-			return nil
+			return false, nil
 		}
 
 		MobilePermissionDeniedCheck(f.G(), err, fmt.Sprintf("%s: %s", f.which, f.filename))
 
 		if os.IsPermission(err) {
 			f.G().Log.Warning("Permission denied opening %s file %s", f.which, f.filename)
-			return nil
+			return true, nil
```

Comment:
It's weird to me that this returns a `nil` error on this case.

Reply:
i'm not sure either, @patrickxb?